Library/Homebrew/cmd/--env.rb

```diff
@@ -3,19 +3,37 @@
 
 require "extend/ENV"
 require "build_environment"
+require "utils/shell"
 
 module Homebrew
+  def explicit_shell
+    shell = ARGV.value("shell")
+    return :noshell if shell == "none"
```

Comment:
Right now `--shell` or `--shell=` is not handled distinctly from the no arguments case (because `ARGV.value("shell")` doesn't distinguish these cases, I think.) and has the original behavior if your shell happens to be `bash`, yes.

`--shell=bash`, however, does not currently preserve the original behavior, since it overrules the interactiveness check.

`--shell=none` is a silly hack to force the `brew --env` to output in `key: value` format. Without `--shell=none` there isn't a way to get the colon format for the environment variables without faking a tty as far as I know. `--shell=none` should probably be renamed.

Just out of curiosity, why does the original behavior of `brew --env` care about whether output is a tty or not? It might make sense to remove that check if `brew --env` is going to take arguments anyway.